### PR TITLE
i1687: Run bb8 tests through TeamCity

### PR DESCRIPTION
**Pull request by MartinEden**
https://vimc.myjetbrains.com/youtrack/issue/VIMC-1687

This doesn't add any new tests. It just gets the tests running through TeamCity on every commit. I say "just", but it's taken me several hours!

Python has gone through several unit testing frameworks in the last few years. Part of the effort has been researching which one to use (looks like `pytest`) and then upgrading things to work with that.

### CODE REVIEW

File: bin/requirements.txt [hunk 1]
```diff
@@ -1,3 +1,2 @@
-python-crontab==2.2.8
```
Comment (MartinEden) [inline bin/requirements.txt:1]: Noticed we didn't need this any more

File: testing/test-setup.sh [hunk 2]
```diff
@@ -1,17 +0,0 @@
...
-    exit 1
-fi
-
-IMAGE_NAME=bb8_setup
-
-docker build --rm -t $IMAGE_NAME $HERE
-docker run --rm -v ${BB8_ROOT}:/src \
-       -e VAULT_AUTH_GITHUB_TOKEN=$VAULT_AUTH_GITHUB_TOKEN \
-       $IMAGE_NAME \
-       /src/setup
```
Comment (MartinEden) [inline testing/test-setup.sh:17]: For now I have just removed this script which tested setting up bb8 on a new machine, as it was out of date, from well before we dockerised bb8. I may write a replacement for this in the future, or I may confine myself to unit tests.

**Review by hillalex — APPROVED** [review]

**Review by richfitz — CHANGES_REQUESTED** [review]
Nice :)

Is it worth setting up with https://github.com/JetBrains/teamcity-messages so that we get easier-to-use test failures too?

**Comment by MartinEden** [comment]
@richfitz Already done, I thought. See `requirements-dev.txt` and the entrypoint of the tests `Dockerfile`.

**Comment by richfitz** [comment]
Sorry - you're right - I must have picked up the wrong build

**Review by richfitz — APPROVED** [review]

**Pull request merged**